morpheus/_lib/include/morpheus/messages/meta.hpp

```diff
@@ -68,6 +69,23 @@ class MessageMeta
      */
     virtual MutableTableInfo get_mutable_info() const;
 
+    /**
+     * @brief Returns true if the underlying DataFrame's index is unique and monotonic. Sliceable indices have better
+     * performance since a range of rows can be specified by a start and stop index instead of requiring boolean masks.
+     *
+     * @return bool
+     */
+    bool has_sliceable_index() const;
+
+    /**
+     * @brief Replaces the index in the underlying dataframe if the existing one is not unique and monotonic. The old
+     * index will be preserved in a column named `_index_{old_index.name}`. If `has_sliceable_index() == true`, this is
+     * a no-op.
+     *
+     * @return std::string The name of the column with the old index or nullopt if no changes were made.
+     */
+    virtual std::optional<std::string> ensure_sliceable_index();
```

Comment:
I'm assuming we're returning optional since the index often doesn't have a name and an empty string often is the name of the index?

Reply:
Not exactly, `ensure_sliceable_index` is something you can just call and its a no-op if there is nothing to do (this means you dont have to call `has_sliceable_index` before calling `ensure_sliceable_index` since we do that anyways). However, if you want to know if something did happen, thats what the optional is for. In the docstring `The name of the column with the old index or nullopt if no changes were made`